products/healthcare/api.yaml

```diff
@@ -540,3 +540,74 @@ objects:
         'Creating a HL7v2 Store':
           'https://cloud.google.com/healthcare/docs/how-tos/hl7v2'
       api: 'https://cloud.google.com/healthcare/docs/reference/rest/v1/projects.locations.datasets.hl7V2Stores'
+  - !ruby/object:Api::Resource
+    name: 'ConsentStore'
+    kind: "healthcare#consentStore"
+    base_url: "{{dataset}}/consentStores?consentStoreId={{name}}"
+    self_link: "{{dataset}}/consentStores/{{name}}"
+    update_verb: :PATCH
+    update_mask: true
+    min_version: beta
+    description: |
+      The Consent Management API is a tool for tracking user consents and the documentation associated with the consents. 
+    iam_policy: !ruby/object:Api::Resource::IamPolicy
+      method_name_separator: ':'
+      fetch_iam_policy_verb: :GET
+      parent_resource_attribute: 'consent_store_id'
+      import_format: ["{{%dataset}}/consentStores/{{name}}", "{{name}}"]
+      base_url: "{{%dataset}}/consentStores/{{name}}"
+      self_link: "{{%dataset}}/consentStores/{{name}}"
+    parameters:
+      - !ruby/object:Api::Type::ResourceRef
+        name: "dataset"
+        description: |
+          Identifies the dataset addressed by this request. Must be in the format
+          'projects/{project}/locations/{location}/datasets/{dataset}'
+        required: true
+        input: true
+        resource: 'Dataset'
+        imports: 'selfLink'
+        url_param_only: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        description: |
+          The last part of the ConsentStore name, for example:
+          "consent1"
+        required: true
+        input: true
+        url_param_only: true
+      - !ruby/object:Api::Type::String
+        name: 'defaultConsentTtl'
+        required: false
+        description: |
+          Default time to live for consents in this store. Must be at least 24 hours. Updating this field will not affect the expiration time of existing consents.
+
+          A duration in seconds with up to nine fractional digits, terminated by 's'. Example: "3.5s".
+      - !ruby/object:Api::Type::Boolean
+        name: 'enableConsentCreateOnUpdate'
+        required: false
+        description: |
+          If true, [consents.patch] [google.cloud.healthcare.v1beta1.consent.UpdateConsent] creates the consent if it does not already exist.
```

Comment:
Why would this do anything? If terraform detects that the resource does not exist, it will attempt to create it via the standard create call.

Reply:
This is a param pass to create a consent store. "creates the consent" is refer to the item in the store, not the consent store.